libs/openchallenges/ui/src/lib/challenge-card/challenge-card.component.ts

```diff
@@ -63,6 +57,48 @@ export class ChallengeCardComponent implements OnInit {
         : this.imageService.getImage({
             objectKey: 'banner-default.svg',
           });
+      if (this.challenge.endDate && this.status === 'completed') {
+        const timeSince = this.calcTimeDiff(this.challenge.endDate);
+        if (timeSince) {
+          this.time_info = `Ended ${timeSince} ago`;
+        }
+      } else if (this.challenge.endDate && this.status === 'active') {
+        this.time_info = `Ends in ${this.calcTimeDiff(this.challenge.endDate)}`;
+      } else if (this.challenge.startDate && this.status === 'upcoming') {
+        this.time_info = `Starts in ${this.calcTimeDiff(
+          this.challenge.startDate
+        )}`;
+      }
     }
   }
+
+  calcTimeDiff(date: string) {
```

Comment:
Declare the return type.

```suggestion
  calcTimeDiff(date: string): string {
```

Reply:
To be addressed in a different PR.

Reply:
@vpchung Please open a ticket with my comment as Description